spec/batch_loader_spec.rb

```diff
@@ -87,6 +87,55 @@
     end
   end
 
+  context 'multi load with meta programming' do
+    it 'loads multi items' do
+      items = []
+      {
+        user: { name: 'User', ids: [1, 2, 3, 4, 5] },
+        post: { name: 'Post', ids: [2, 3, 4, 1, 2] },
+      }.each do |type, kind|
+        kind[:ids].map do |id|
+          items << BatchLoader.for(id, type).batch do |ids, loader, slug|
```

Comment:
I think that these batch blocks in the tests have a `source_location` something like `["spec/batch_loader_spec.rb", 98]`. Perhaps, to add a test where `source_location` doesn't work correctly we need to use something like `eval`.